src/Chapter4.hs

```diff
@@ -673,6 +712,48 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a = None | Node { nodeValue :: a
+                          , nodeLeft :: Tree a
+                          , nodeRight :: Tree a } deriving (Show)
```

Comment:
It is antipattern in Haskell to have named arguments in the sum types as they create partial functions that would fail in the runtime. For example, `nodeValue` has type `Tree a -> a`. You can perfectly do the following, and type system won't stop you:
```haskell
nodeValue None
```

To resolve it, you have 2 options:

1. Move named record into a separate record type and use it in the sum type
2. Use unnamed records

Reply:
Thanks!